src/components/App/index.js

```diff
@@ -1,17 +1,102 @@
-import React from "react";
-import { Segment, Divider, Header, Icon } from "semantic-ui-react";
+import React, { useState } from "react";
+import {
+  Segment,
+  Divider,
+  Header,
+  Icon,
+  Button,
+  Modal,
+  Transition,
+  Label,
+} from "semantic-ui-react";
 import "semantic-ui-css/semantic.min.css";
 import roadmapData from "../../data/roadmapData.json";
 import SingleTree from "../SingleTree";
 
 const App = () => {
+  const [state, setState] = useState({
+    showInstruction:
+      localStorage.getItem("instructionViewedAR") === "true" ? false : true,
+  });
+
+  const openInstruction = () =>
+    setState((prevState) => ({
+      ...prevState,
+      showInstruction: true,
+    }));
+  const handleCloseInstruction = () => {
+    localStorage.setItem("instructionViewedAR", true);
+    setState((prevState) => ({
+      ...prevState,
+      showInstruction: false,
+    }));
+  };
+
```

Comment:
I am not react master but wouldn't it be better to create a `useState` hook purely to handle modal being open. I think it would make the code easier and shorter. It would look something like this:
```
const [isInstructionOpened, setIsInstructionOpened] = useState(localStorage.getItem("instructionViewedAR") === "true" ? false : true)

const handleInstructionModal = () => setIsInstructionOpened(!isInstructionOpened)
```

Reply:
absolutely right, as we not have any other variables in state, we can simplify it. thanks!